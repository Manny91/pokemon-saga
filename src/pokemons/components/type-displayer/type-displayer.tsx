import React from "react";
import styled from "../../../styled-components";
import { PokemonType } from "../../../services/pokemon.service";

interface PokemonTypeProps {
  types: PokemonType[];
}
export function PokemonTypeDisplayer({ types }: PokemonTypeProps) {
  return types ? (
    <TypeWrapper>
      {types.map((pokeType: PokemonType, index) => (
        <PokemonTypeBanner className={pokeType.type.name} key={index}>
          {pokeType.type.name}
        </PokemonTypeBanner>
      ))}
    </TypeWrapper>
  ) : null;
}
const TypeWrapper = styled.div`
  background: linear-gradient(14deg, #afafaf 50%, #ffffff 80%, #afafaf 90%);
  border: groove ${props => props.theme.colors.greenDarkBorder} 3px;
  width: 100%;
`;
const PokemonTypeBannerHeader = styled.div`
  height: 35px;
  text-transform: uppercase;
  text-align: center;
  border: groove ${props => props.theme.colors.blackBorder} 3px;
  justify-content: center;
  display: flex;
  align-items: center;
  font-size: 19px;
  font-weight: bold;
`;

const PokemonTypeBanner = styled(PokemonTypeBannerHeader)`
  border-top-right-radius: 3px;
  border-top-left-radius: 3px;
  border-bottom-right-radius: 3px;
  border-bottom-left-radius: 3px;
  margin: 3px;

  &.normal {
    background: ${props => props.theme.colors.pokemonTypes.normal};
  }
  &.fighting {
    background: ${props => props.theme.colors.pokemonTypes.fighting};
  }
  &.flying {
    background: ${props => props.theme.colors.pokemonTypes.flying};
  }
  &.poison {
    background: ${props => props.theme.colors.pokemonTypes.poison};
  }
  &.ground {
    background: ${props => props.theme.colors.pokemonTypes.ground};
  }
  &.rock {
    background: ${props => props.theme.colors.pokemonTypes.rock};
  }
  &.bug {
    background: ${props => props.theme.colors.pokemonTypes.bug};
  }
  &.ghost {
    background: ${props => props.theme.colors.pokemonTypes.ghost};
  }
  &.steel {
    background: ${props => props.theme.colors.pokemonTypes.steel};
  }
  &.fire {
    background: ${props => props.theme.colors.pokemonTypes.fire};
  }
  &.water {
    background: ${props => props.theme.colors.pokemonTypes.water};
  }
  &.grass {
    background: ${props => props.theme.colors.pokemonTypes.grass};
  }
  &.electric {
    background: ${props => props.theme.colors.pokemonTypes.electric};
  }
  &.psychic {
    background: ${props => props.theme.colors.pokemonTypes.psychic};
  }
  &.ice {
    background: ${props => props.theme.colors.pokemonTypes.ice};
  }
  &.dragon {
    background: ${props => props.theme.colors.pokemonTypes.dragon};
  }
  &.dark {
    background: ${props => props.theme.colors.pokemonTypes.dark};
  }
  &.fairy {
    background: ${props => props.theme.colors.pokemonTypes.fairy};
  }
  &.unknown {
    background: ${props => props.theme.colors.pokemonTypes.unknown};
  }
  &.shadow {
    background: ${props => props.theme.colors.pokemonTypes.shadow};
  }
`;
