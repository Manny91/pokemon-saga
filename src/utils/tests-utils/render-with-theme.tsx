import React from "react";
import { ThemeProvider } from "../../styled-components";
import { theme } from "../../theme";
import { render } from "@testing-library/react";

export const renderWithTheme = (ui: JSX.Element) => {
  return render(<ThemeProvider theme={theme}>{ui}</ThemeProvider>);
};
