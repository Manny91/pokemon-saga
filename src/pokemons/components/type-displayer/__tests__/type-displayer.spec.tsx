import React from "react";
import { PokemonTypeDisplayer } from "../type-displayer";
import "@testing-library/jest-dom/extend-expect";
import "jest-styled-components";
import { renderWithTheme } from "../../../../utils/tests-utils/render-with-theme";

describe("PokemonTypeDisplayer Component", () => {
  it("should render properly", () => {
    const testProps = {
      types: [{ slot: 0, type: { name: "grass" } }]
    };
    const subject = renderWithTheme(<PokemonTypeDisplayer {...testProps} />);
    const tree = subject.container;
    expect(tree).toMatchSnapshot();
  });
});
