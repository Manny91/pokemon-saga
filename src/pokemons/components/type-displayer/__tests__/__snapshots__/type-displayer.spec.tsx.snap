// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonTypeDisplayer Component should render properly 1`] = `
.c0 {
  background: linear-gradient(14deg,#afafaf 50%,#ffffff 80%,#afafaf 90%);
  border: groove #4e4e4e 3px;
  width: 100%;
}

.c1 {
  height: 35px;
  text-transform: uppercase;
  text-align: center;
  border: groove #757575 3px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 19px;
  font-weight: bold;
  border-top-right-radius: 3px;
  border-top-left-radius: 3px;
  border-bottom-right-radius: 3px;
  border-bottom-left-radius: 3px;
  margin: 3px;
}

.c1.normal {
  background: #bfbfbf;
}

.c1.fighting {
  background: #d87c58;
}

.c1.flying {
  background: #999ade;
}

.c1.poison {
  background: #925192;
}

.c1.ground {
  background: #dea761;
}

.c1.rock {
  background: #897864;
}

.c1.bug {
  background: #b1c967;
}

.c1.ghost {
  background: #c195dc;
}

.c1.steel {
  background: #49769c;
}

.c1.fire {
  background: #cf1414;
}

.c1.water {
  background: #1689de;
}

.c1.grass {
  background: #47a047;
}

.c1.electric {
  background: #e6b700;
}

.c1.psychic {
  background: #fa43b8;
}

.c1.ice {
  background: #98c3de;
}

.c1.dragon {
  background: #89315d;
}

.c1.dark {
  background: #282433;
}

.c1.fairy {
  background: #dca0ce;
}

.c1.unknown {
  background: #545454;
}

.c1.shadow {
  background: #364163;
}

<div>
  <div
    class="c0"
  >
    <div
      class="sc-AxirZ c1 grass"
    >
      grass
    </div>
  </div>
</div>
`;
